plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -1312,59 +1314,45 @@ private ConnectorMaterializedViewDefinition createMaterializedViewDefinition(
-                // Retry, as we're catching broadly.
-                throw new MaterializedViewMayBeBeingRemovedException(e);
-            }
-
             String viewOriginalText = table.getViewOriginalText();
             if (viewOriginalText == null) {
                 throw new TrinoException(ICEBERG_BAD_DATA, "Materialized view did not have original text " + viewName);
             }
             return getMaterializedViewDefinition(
-                    icebergTable,
                     Optional.ofNullable(table.getOwner()),
                     viewOriginalText,
                     storageTableName);
         }
 
         SchemaTableName storageTableName = new SchemaTableName(viewName.getSchemaName(), tableNameWithType(viewName.getTableName(), MATERIALIZED_VIEW_STORAGE));
-        Table icebergTable;
-        try {
-            TableMetadata metadata = getMaterializedViewTableMetadata(session, storageTableName, storageMetadataLocation);
-            IcebergTableOperations operations = tableOperationsProvider.createTableOperations(
-                    this,
-                    session,
-                    storageTableName.getSchemaName(),
-                    storageTableName.getTableName(),
-                    Optional.empty(),
-                    Optional.empty());
-            operations.initializeFromMetadata(metadata);
-            icebergTable = new BaseTable(operations, quotedTableName(storageTableName), TRINO_METRICS_REPORTER);
-        }
-        catch (RuntimeException e) {
-            // The materialized view could be removed concurrently. This may manifest in a number of ways, e.g.
-            // - org.apache.iceberg.exceptions.NotFoundException when accessing manifest file
-            // - other failures when reading storage table's metadata files
-            // Retry, as we're catching broadly.
-            throw new MaterializedViewMayBeBeingRemovedException(e);
-        }
-
         return getMaterializedViewDefinition(
-                icebergTable,
                 Optional.ofNullable(table.getOwner()),
                 table.getViewOriginalText(),
                 storageTableName);
     }
 
+    @Override
+    public Map<String, Object> getMaterializedViewProperties(ConnectorSession session, SchemaTableName viewName, ConnectorMaterializedViewDefinition definition)
+    {
+        if (definition.getStorageTable().isEmpty()) {
+            throw new TrinoException(ICEBERG_INVALID_METADATA, "Materialized view definition is missing a storage table");
+        }
+
+        SchemaTableName storageTableName = definition.getStorageTable().get().getSchemaTableName();
+        try {
+            Table storageTable = loadTable(session, definition.getStorageTable().orElseThrow().getSchemaTableName());
+            return ImmutableMap.<String, Object>builder()
+                    .putAll(getIcebergTableProperties(storageTable))
+                    .put(STORAGE_SCHEMA, storageTableName.getSchemaName())
```

Comment:
Should this be populated only when different from MV name's schema?
or is it pre-existing?

Reply:
That's pre-existing from https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java#L441

I'm not sure if it's required or not